nekoyume/util.py

```diff
@@ -1,7 +1,24 @@
+import functools
+from typing import Callable
+
 from coincurve import PublicKey
 from keccak import sha3_256
 
+from nekoyume.exc import InvalidMoveError
+
 
 def get_address(public_key: PublicKey) -> str:
     """Derive an Ethereum-style address from the given public key."""
     return '0x' + sha3_256(public_key.format(False)[1:]).hexdigest()[-40:]
+
+
+def ensure_block(f: Callable) -> Callable:
+    @functools.wraps(f)
+    def decorator(self, *args, **kwargs):
+        try:
+            if self.block:
+                return f(self, *args, **kwargs)
+        except AttributeError:
+            raise InvalidMoveError
```

Comment:
Since `AttributeError` raised by `f(...)` shouldn't be caught (but only `self.block`) here, it should be like:

```python
try:
    block = self.block
except AttributeError:
    block = None
if block:
    return f(self, *args, **kwargs)
```